src/runtime/moduleobject.cs

```diff
@@ -320,6 +349,25 @@ protected override void Clear()
             base.Clear();
         }
 
+        /// <summary>
+        /// Override the setattr implementation.
+        /// This is needed because the import mechanics need
+        /// to set a few attributes
+        /// </summary>
+        [ForbidPythonThreads]
```

Comment:
Why do you have to forbid threads here?

Reply:
Because there's a call to `Runtime.PyDict_SetItem`. I might be missing something and this is always called from CPython and the GIL is guaranteed to be held.